frontend/pages/policies/ManagePoliciesPage/components/RunScriptModal/RunScriptModal.tsx

```diff
@@ -0,0 +1,241 @@
+import React, { useCallback, useState } from "react";
+import { useQuery } from "react-query";
+import { omit } from "lodash";
+
+import { DEFAULT_USE_QUERY_OPTIONS } from "utilities/constants";
+
+import scriptsAPI, {
+  IListScriptsQueryKey,
+  IScriptsResponse,
+} from "services/entities/scripts";
+
+import { IPolicyStats } from "interfaces/policy";
+import { IScript } from "interfaces/script";
+
+// @ts-ignore
+import Dropdown from "components/forms/fields/Dropdown";
+import Checkbox from "components/forms/fields/Checkbox";
+import TooltipTruncatedText from "components/TooltipTruncatedText";
+import DataError from "components/DataError";
+import Spinner from "components/Spinner";
+import CustomLink from "components/CustomLink";
+import Button from "components/buttons/Button";
+import Modal from "components/Modal";
+
+// name avoids class name conflicts with hosts/details/HostDetailsPage/modals/RunScriptModal
+const baseClass = "policy-run-script-modal";
+
+interface IScriptDropdownField {
+  name: string;
+  value: number;
+}
+
+interface IFormPolicy {
+  name: string;
+  id: number;
+  runScriptEnabled: boolean;
+  scriptIdToRun?: number;
+}
+
+export type IPolicyRunScriptFormData = IFormPolicy[];
+
+interface IRunScriptModal {
+  onExit: () => void;
+  onSubmit: (formData: IPolicyRunScriptFormData) => void;
+  isUpdating: boolean;
+  policies: IPolicyStats[];
+  teamId: number;
+}
+
+const RunScriptModal = ({
+  onExit,
+  onSubmit,
+  isUpdating,
+  policies,
+  teamId,
+}: IRunScriptModal) => {
+  const [formData, setFormData] = useState<IPolicyRunScriptFormData>(
+    policies.map((policy) => ({
+      name: policy.name,
+      id: policy.id,
+      runScriptEnabled: !!policy.run_script,
+      scriptIdToRun: policy.run_script?.id,
+    }))
+  );
+
+  const anyEnabledWithoutSelection = formData.some(
+    (policy) => policy.runScriptEnabled && !policy.scriptIdToRun
+  );
+
+  const {
+    data: availableScripts,
+    isLoading: isLoadingAvailableScripts,
+    isError: isAvailableScriptsError,
+  } = useQuery<IScriptsResponse, Error, IScript[], [IListScriptsQueryKey]>(
+    [
+      {
+        scope: "scripts",
+        team_id: teamId,
+      },
+    ],
+    ({ queryKey: [queryKey] }) =>
+      scriptsAPI.getScripts(omit(queryKey, "scope")),
+    {
+      select: (data) => data.scripts,
+      ...DEFAULT_USE_QUERY_OPTIONS,
+    }
+  );
+
+  const onUpdate = useCallback(() => {
+    onSubmit(formData);
+  }, [formData, onSubmit]);
+
+  const onChangeEnableRunScript = useCallback(
+    (newVal: { policyName: string; value: boolean }) => {
```

Comment:
Feels like we should be passing a policy object around including ID, since we'll always have the PK by this point? Matching on name when we don't have to match on name makes me nervouse 😅

Reply:
Hmm, that's a good call. Will look into it.